src/index.ts

```diff
@@ -52,19 +41,29 @@ export const useSubjectValue = <T, R>(
       }
     };
 
-    return [update, unsubscribe];
+    return [update, input$];
   }, []);
 
-  // Set active flag to true while updating and call it with new input
-  state.current.active = true;
-  update(input);
-  state.current.active = false;
+  useIsomorphicEffect(() => {
+    const [unsubscribe] = pipe(
+      fn(input$),
+      subscribe(forceUpdate)
+    );
+
+    return unsubscribe;
+  }, []);
 
-  // Let React call unsubscribe on unmount
-  useEffect(() => unsubscribe, []);
+  useIsomorphicEffect(() => {
```

Comment:
`useLayoutEffect` runs after mounting (“did mount”) and wouldn’t yield the correct value on initial mounts or on updates that are synchronous

Reply:
This is the way React intends to do it as far as I know, we could introduce another step to see if there's a value to introduce synchronousity as in `useState(() => getSomeValue)`